packages/riipen-ui/src/components/Chip.test.jsx

```diff
@@ -0,0 +1,136 @@
+import toJson from "enzyme-to-json";
+import React from "react";
+
+import Chip from "./Chip";
+
+it("renders without errors", () => {
+  try {
+    mount(<Chip />);
+  } catch (error) {
+    expect(true).toEqual(false);
+  }
+});
+
+describe("default props", () => {
+  it("sets correct default props", () => {
+    const defaultProps = new Chip().type.defaultProps;
+    defaultProps.classes.push("riipen-chip", "riipen");
+
+    const wrapper = mount(<Chip />);
+
+    const component = wrapper.find("Chip").childAt(0);
+    expect(component.name()).toEqual("div");
+    expect(component.hasClass(defaultProps.color)).toEqual(true);
+    expect(component.hasClass("disabled")).toEqual(false);
+    expect(component.hasClass("hover")).toEqual(false);
+    expect(component.hasClass(defaultProps.size)).toEqual(true);
+    expect(component.hasClass(defaultProps.variant)).toEqual(true);
+    expect(component.childAt(0).name()).toEqual("span");
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+});
+
+describe("variant prop", () => {
+  it("outlines chip given outlined variant", () => {
+    const wrapper = mount(<Chip variant="outlined" />);
+
+    expect(
+      wrapper
+        .find("Chip")
+        .childAt(0)
+        .hasClass("outlined")
+    ).toEqual(true);
+  });
+});
+
+describe("component prop", () => {
+  it("component can be any element", () => {
+    const wrapper = mount(<Chip component="a" />);
+
+    expect(wrapper.find("a")).toHaveLength(1);
+    expect(
+      wrapper
+        .find("Chip")
+        .childAt(0)
+        .name()
+    ).toEqual("a");
+  });
+});
+
+describe("color prop", () => {
+  it("sets custom color", () => {
+    const custom = "dark";
+
+    const wrapper = mount(<Chip color={custom} />);
+
+    expect(wrapper.find("Chip").props().color).toEqual(custom);
+    expect(
+      wrapper
+        .find("Chip")
+        .childAt(0)
+        .hasClass(custom)
+    ).toEqual(true);
+  });
+
+  it("gives an error when given an invalid color", () => {
+    const errors = jest.spyOn(console, "error").mockImplementation();
+
+    mount(<Chip color="red" />);
+
+    expect(errors).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe("label prop", () => {
+  it("displays label given as the label prop", () => {
+    const label = "A test label";
+
+    const wrapper = mount(<Chip label={label} />);
+
+    expect(wrapper.find("Chip").props().label).toEqual(label);
+    expect(
+      wrapper
+        .find("Chip")
+        .find("span")
+        .text()
+    ).toEqual(label);
+  });
+
+  it("displays label from child", () => {
+    const label = "A label given as a child";
+    const child = <div>{label}</div>;
+
+    const wrapper = mount(<Chip>{child}</Chip>);
+
+    expect(
+      wrapper
+        .find(".label")
+        .children()
+        .contains(child)
+    ).toEqual(true);
+    expect(
+      wrapper
+        .find(".label")
+        .childAt(0)
+        .text()
+    ).toEqual(label);
+  });
```

Comment:
I don't think this test is quite right.

Looking at the source code for `<Chip>`, it states:

```js
const label = this.props.label || this.props.children;
```

Meaning that:

1. if a label is provided, it uses that
2. if children are provided, it uses that
3. if label and children are provided it uses the label

So I think our tests should probably reflect that.

We should actually change this test to `"prioritizes displaying label over children"` where was pass both label and children into the props and make sure only label shows up.

Then, we should actually make another `describe('children prop' () => {...});` which tests `"displays child"` or something.